flaky test: TestTimeoutWhenAggregationKeepHappen

### Component(s)

pkg/stanza

### Describe the issue you're reporting


https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/4676595428

```
ok  	github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/operator/transformer/noop	0.023s
--- FAIL: TestTimeoutWhenAggregationKeepHappen (0.10s)
    recombine_test.go:507: 
        	Error Trace:	/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/pkg/stanza/operator/transformer/recombine/recombine_test.go:507
        	Error:      	Not equal: 
        	            	expected: "startnext"
        	            	actual  : "startnextnext"
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1 +1 @@
        	            	-startnext
        	            	+startnextnext
        	Test:       	TestTimeoutWhenAggregationKeepHappen
FAIL
FAIL	github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/operator/transformer/recombine	0.139s
```

--- a/pkg/stanza/operator/transformer/recombine/recombine_test.go
+++ b/pkg/stanza/operator/transformer/recombine/recombine_test.go
@@ -566,7 +566,8 @@ func TestTimeoutWhenAggregationKeepHappen(t *testing.T) {
 	require.NoError(t, recombine.Process(ctx, e))
 
 	go func() {
-		next := e.Copy()
+		next := entry.New()
+		next.Timestamp = time.Now()
 		next.Body = "next"
 		for {
 			time.Sleep(cfg.ForceFlushTimeout / 2)

